Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +182,25 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)maybeConsumeBundlePaths:(NSDictionary<NSAttributedStringDocumentReadingOptionKey, id> *)options
+{
+    auto* readAccessPaths = dynamic_objc_cast<NSArray<NSURL *>>(options[WKReadAccessFileURLsOption]);
```

Comment:
You should know that the `dynamic_objc_cast<>` here only checks that the outer object is an `NSArray` (or `NSMutableAray`).

It does not check that the contents of the `NSArray` are only `NSURL` objects.

Objective-C generics are merely syntactic sugar to catch mistakes during compiling, but are never type-checked at runtime.

Reply:
Ah, okay. I think it is sufficient to validate on setting the value (as you propose below).